src/ape_pm/projects.py

```diff
@@ -101,3 +109,45 @@ def extract_config(self, **overrides) -> ApeConfig:
 
         model = {**migrated_config_data, **overrides}
         return ApeConfig.model_validate(model)
+
+
+class FoundryProject(ProjectAPI):
+    """
+    Helps Ape read configurations from foundry projects
+    and lessens the need of specifying ``config_override:``
+    for foundry-based dependencies.
+    """
+
+    @property
+    def foundry_config_file(self) -> Path:
+        return self.path / "foundry.toml"
+
+    @property
+    def is_valid(self) -> bool:
+        return self.foundry_config_file.is_file()
+
+    def extract_config(self, **overrides) -> "ApeConfig":
+        ape_cfg: dict = {}
+        data = tomllib.loads(self.foundry_config_file.read_text())
+        profile = data.get("profile", {})
+        root_data = profile.get("default", {})
+
+        # Handle root project configuration.
+        # NOTE: The default contracts folder name is `src` in foundry
+        #  instead of `contracts`, hence the default.
+        ape_cfg["contracts_folder"] = root_data.get("src", "src")
+
+        # Handle all ape-solidity configuration.
+        solidity_data: dict = {}
+        if solc_version := (root_data.get("solc") or root_data.get("solc_version")):
+            solidity_data["version"] = solc_version
+        if remappings := root_data.get("remappings"):
```

Comment:
- [ ] make this actually work, coordinating with foundry's lib dir and ape's dependencies cfg....